src/test-provider/test-provider-helper.ts

```diff
@@ -191,11 +191,8 @@ export class JestTestRun implements JestExtOutput, TestRunProtocol {
       return this.options.end();
     }
 
-    if (this.parentRun) {
-      this.parentRun.end();
```

Comment:
This line seems to break a lot of the tests. Not sure if it is a real problem though, or it can be ignored. Also not sure how to solve it.

Reply:
Is this change just a refactoring? I noticed the refactored logic is not the same as the original. Is there any other reason for changing the logic?

Reply:
> Is this change just a refactoring? I noticed the refactored logic is not the same as the original. Is there any other reason for changing the logic?

I actually added a comment on this line myself on the pull request.

It's what makes the tests fail around remembering to end the runs.

However, with this line on, when one of the tests switch status to green, the rest of them will also go out of the "running spinner" state, although they are still running.

Do you have any idea of why this happens?

Reply:
> However, with this line on, when one of the tests switch status to green, the rest of them will also go out of the "running spinner" state, although they are still running.

My guess (without actually tracing the calls) is the code assumed batch mode, so once an "end" event is received, it will try to close the run. The TestExplorer will then mark all items "done" in that run accordingly. You might need to introduce a new stage/event when processing the partial result, or at least not trigger the "end" event before the whole run actually finishes...